stl/inc/syncstream

```diff
@@ -218,11 +218,12 @@ protected:
         }
 
         const _Size_type _New_capacity  = _Calculate_growth(_Buf_size, _Buf_size + 1, _Max_allocation);
-        const _Elem* const _Old_ptr     = streambuf_type::pbase();
+        _Elem* const _Old_ptr           = streambuf_type::pbase();
         const _Size_type _Old_data_size = _Get_data_size();
 
         _Elem* const _New_ptr = _Unfancy(_Al.allocate(_New_capacity));
         _Traits::copy(_New_ptr, _Old_ptr, _Old_data_size);
+        _Al.deallocate(_Refancy<_Pointer>(_Old_ptr), _Old_data_size);

```

Comment:
I suspect that this is incorrect. In this function, we have (omitting irrelevant lines):

```cpp
const _Size_type _Buf_size       = _Get_buffer_size();
// ...
_Elem* const _Old_ptr           = streambuf_type::pbase();
const _Size_type _Old_data_size = _Get_data_size();
```

`_Buf_size` is the old *capacity*, while `_Old_data_size` is the old *size*. The previous line `_Traits::copy(_New_ptr, _Old_ptr, _Old_data_size);` is correctly copying the old size, but when we deallocate, we need to use the old capacity. (If we have a guarantee here that the old capacity and old size are identical, we should still use the old capacity for clarity.)

Observe what `_Tidy()` does:

https://github.com/microsoft/STL/blob/17fde2cbab6e8724d81c9555237c9a623d7fb954/stl/inc/syncstream#L241-L245

To be careful, I think we should do the same thing (including the nonzero test).

Aside (pre-existing and too much to change here): This illustrates the danger of using the word "size" for very different concepts. We should be disciplined about "size" versus "capacity" terminology.

Reply:
I think it was correct because we call `overflow` only in two places and only when the buffer has 0 available bytes so capacity == size
https://github.com/microsoft/STL/blob/17fde2cbab6e8724d81c9555237c9a623d7fb954/stl/inc/streambuf#L166-L168
https://github.com/microsoft/STL/blob/17fde2cbab6e8724d81c9555237c9a623d7fb954/stl/inc/streambuf#L336-L346
But I agree that do the same thing like `_Tidy` is better